pyretailscience/plots/line.py

```diff
@@ -0,0 +1,114 @@
+
+It focuses on  visualizing sequences that resemble time-based data, such as "days since an event" or "months since a
+competitor opened." However, it does not explicitly handle datetime values. For actual time-based plots using
+datetime objects, please refer to the **`timeline`** module.
+
+The sequences used in this module can include values like "days since an event" (e.g., -2, -1, 0, 1, 2) or "months
+since a competitor store opened." **This module is not intended for use with actual datetime values**. If a datetime
+or datetime-like column is passed as **`x_col`**, a warning will be triggered, suggesting the use of the
+**`timeline`** module.
+
+### Core Features
+
+- **Plotting Sequences or Indexes**: Plot one or more value columns (**`value_col`**) with support for sequences like
+-2, -1, 0, 1, 2 (e.g., months since an event), using either the index or a specified x-axis column (**`x_col`**).
+- **Custom X-Axis or Index**: Use any column as the x-axis (**`x_col`**) or plot based on the index if no x-axis column is specified.
+- **Multiple Lines**: Create separate lines for each unique value in **`group_col`** (e.g., categories or product types).
+- **Comprehensive Customization**: Easily customize plot titles, axis labels, and legends, with the option to move the legend outside the plot.
+- **Pre-Aggregated Data**: The data must be pre-aggregated before plotting, as no aggregation occurs within the module.
+
+### Use Cases
+
+- **Daily Trends**: Plot trends such as daily revenue or user activity, for example, tracking revenue since the start of the year.
+- **Event Impact**: Visualize how metrics (e.g., revenue, sales, or traffic) change before and after an important event, such as a competitor store opening or a product launch.
+- **Category Comparison**: Compare metrics across multiple categories over time, for example, tracking total revenue for the top categories before and after an event like the introduction of a new competitor.
+
+### Limitations and Handling of Temporal Data
+
+- **Limited Handling of Temporal Data**: This module can plot simple time-based sequences, such as "days since an event," but it cannot manipulate or directly handle datetime or date-like columns. It is not optimized for actual datetime values.
+If a datetime column is passed or more complex temporal plotting is needed, a warning will suggest using the **`timeline`** module, which is specifically designed for working with temporal data and performing time-based manipulation.
+- **Pre-Aggregated Data Required**: The module does not perform any data aggregation, so all data must be pre-aggregated before being passed in for plotting.
+
+"""
+
+import warnings
+
+import pandas as pd
+from matplotlib.axes import Axes, SubplotBase
+
+import pyretailscience.style.graph_utils as gu
+from pyretailscience.style.tailwind import get_base_cmap
+
+
+def plot(
+    df: pd.DataFrame,
+    value_col: str | list[str],
+    x_label: str,
+    y_label: str,
+    title: str,
```

Comment:
These should all be optional. If they user doesn't supply them then the relevant lable/title should be hidden.
```suggestion
    x_label: str | None = None,
    y_label: str | None = None,
    title: str | None = None,
```